src/options/index.ts

```diff
@@ -67,12 +67,28 @@ urlForm.addEventListener("submit", (e) => {
     return alert("You can't add a website while the focus mode is running.");
   if (!urlInput.value) return alert("Please enter a URL.");
 
-  blockList.push(urlInput.value);
-  browser.storage.local.set({ blockList });
-  addUrlListElement(urlInput.value);
+  if (!isValidURL(urlInput.value)) return alert("Please enter a valid URL.") // verify URL
+
+  if (checkUrlListElements(urlInput.value)) {
+      // if url exists in website-list div, return alert
+      return alert("This URL already exists")
+  } else {
+    // if url doesnt exist in website-list div, add new element to
+    blockList.push(urlInput.value);
+    browser.storage.local.set({ blockList });
+    addUrlListElement(urlInput.value);
+  }
+
   urlInput.value = "";
 });
 
+
+function isValidURL(url: string) {
+  const regex = /^(https?|ftp):\/\/[^\s\/$.?#].[^\s]*$/
```

Comment:
It's good that there is a check if the url is valid, but only accepting url's with "https" or "http" removes the dynamic of simply add an "instagram.com" url, which should also work.